zipline/pipeline/term.py

```diff
@@ -516,6 +585,31 @@ def postprocess(self, data):
         """
         return data
 
+    def _downsampled_type(self):
+        """
+        The expression type to return from self.downsample().
+        """
+        raise NotImplementedError(
+            "downsampling is not yet implemented "
+            "for instances of %s." % type(self).__name__
+        )
+
+    def downsample(self, frequency):
+        """
+        Make a term that computes from ``self`` at lower-than-daily frequency.
+
+        Parameters
+        ----------
+        frequency : str, {'Y', 'Q', 'M', 'W'}
+            A string indicating the desired sampling rate.
+            'Y' -> sample on the first trading day of each calendar year
```

Comment:
@twiecki @llllllllll my other concern with using the pandas conventions is that I think calculating at the start of a period will be the more common desired outcome, but the more obvious period names of `'M'`, `'Y'` and `'Q'` mean all mean the end of the period.  Pandas also doesn't have a start/end distinction for weekly frequency.  Given that I think relatively few of our users are going to know the pandas symbols by heart, I don't think there's a ton to be gained by copying their conventions.  At the same time, there are probably enough users who are familiar with pandas that they'd be confused if we re-used `'M'` and `'Q'` but had them mean different things.

In light of all of the above, what do you think about just using slightly-more-verbose strings like `'month_start'`, `'year_start'`, `'quarter_end'`?